electron_app/electron.js

```diff
@@ -109,34 +109,51 @@ Main._initializeApplication = function () {
     electron.ipcMain.on('app_ready', async(event) => {
       autoUpdater.autoDownload = false;
 
-      autoUpdater.checkForUpdates();
+      autoUpdater.checkForUpdates().then((result) => {
+        console.log(result.updateInfo.version);
 
-      const currentVersion = electron.app.getVersion();
-      const currentVersionIsPrerelease = prereleaseRegex.test(currentVersion);
-      autoUpdater.allowPrerelease = currentVersionIsPrerelease;
 
-      const downloadCancellationToken = new CancellationToken();
+        const currentVersion = electron.app.getVersion();
+        const currentVersionIsPrerelease = prereleaseRegex.test(currentVersion);
+        autoUpdater.allowPrerelease = currentVersionIsPrerelease;
 
-      console.log(`CurrentVersion: ${currentVersion}, CurrentVersionIsPrerelease: ${currentVersionIsPrerelease}`);
+        const downloadCancellationToken = new CancellationToken();
 
-      autoUpdater.addListener('error', (error) => {
-        event.sender.send('update_error');
-      });
+        console.log(`CurrentVersion: ${currentVersion}, CurrentVersionIsPrerelease: ${currentVersionIsPrerelease}`);
+
+        autoUpdater.addListener('error', (error) => {
+          event.sender.send('update_error');
+        });
+
+        autoUpdater.addListener('update-available', (info) => {
+          event.sender.send('update_available', result.updateInfo.version);
 
-      autoUpdater.addListener('update-available', (info) => {
-        event.sender.send('update_available');
+          electron.ipcMain.on('download_update', (event) => {
```

Comment:
This is the shadowing I mentioned earlier.

It could stay as it is, if the other `event` is renamed like I suggested.

Or you could just remove that variable from the callback-signature, since it's never used anyway.

```suggestion
          electron.ipcMain.on('download_update', () => {
```